sphinx/screens-detail/payments/payment-template/payment-template/src/main/java/chat/sphinx/payment_template/ui/PaymentTemplateViewModel.kt

```diff
@@ -0,0 +1,138 @@
+import dagger.hilt.android.lifecycle.HiltViewModel
+import io.matthewnelson.android_feature_navigation.util.navArgs
+import io.matthewnelson.android_feature_viewmodel.SideEffectViewModel
+import io.matthewnelson.android_feature_viewmodel.submitSideEffect
+import io.matthewnelson.concept_coroutines.CoroutineDispatchers
+import io.matthewnelson.concept_views.viewstate.ViewStateContainer
+import io.matthewnelson.concept_views.viewstate.value
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+internal inline val PaymentTemplateFragmentArgs.chatId: ChatId?
+    get() = if (argChatId == ChatId.NULL_CHAT_ID.toLong()) {
+        null
+    } else {
+        ChatId(argChatId)
+    }
+
+internal inline val PaymentTemplateFragmentArgs.contactId: ContactId
+    get() = ContactId(argContactId)
+
+internal inline val PaymentTemplateFragmentArgs.amount: Sat
+    get() = Sat(argAmount)
+
+internal inline val PaymentTemplateFragmentArgs.message: String?
+    get() = if (argMessage.isEmpty()) {
+        null
+    } else {
+        argMessage
+    }
+
+@HiltViewModel
+internal class PaymentTemplateViewModel @Inject constructor(
+    dispatchers: CoroutineDispatchers,
+    savedStateHandle: SavedStateHandle,
+    private val app: Application,
+    private val messageRepository: MessageRepository,
+    val navigator: PaymentTemplateNavigator
+): SideEffectViewModel<
+        Context,
+        PaymentTemplateSideEffect,
+        PaymentTemplateViewState>(dispatchers, PaymentTemplateViewState.Idle)
+{
+
+    private val sendPaymentBuilder = SendPayment.Builder()
+
+    private val args: PaymentTemplateFragmentArgs by savedStateHandle.navArgs()
+
+    val templateImagesViewStateContainer: ViewStateContainer<TemplateImagesViewState> by lazy {
+        ViewStateContainer(TemplateImagesViewState.LoadingTemplateImages)
+    }
+
+    val selectedTemplateViewStateContainer: ViewStateContainer<SelectedTemplateViewState> by lazy {
+        ViewStateContainer(SelectedTemplateViewState.Idle)
+    }
+
+    suspend fun loadTemplateImages() {
+        templateImagesViewStateContainer.updateViewState(TemplateImagesViewState.LoadingTemplateImages)
+
+        when (val response = messageRepository.getPaymentTemplates()) {
+            is Response.Error -> {
+                templateImagesViewStateContainer.updateViewState(
+                    TemplateImagesViewState.TemplateImages(listOf())
+                )
+            }
+            is Response.Success -> {
+                templateImagesViewStateContainer.updateViewState(
+                    TemplateImagesViewState.TemplateImages(response.value)
+                )
+            }
+        }
+    }
+
+    fun selectTemplate(position: Int) {
+        if (templateImagesViewStateContainer.value is TemplateImagesViewState.TemplateImages) {
+
+            (templateImagesViewStateContainer.value as TemplateImagesViewState.TemplateImages)
```

Comment:
use local value to store static reference to VScontainer value. This can potentially throw ClassCastException if value is changed.